src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1398,6 +1455,104 @@ private MqttServerStream resolvePublisher(
             return stream;
         }
 
+        private void resolveSession(
+            long traceId,
+            long authorization,
+            int flags,
+            int willDelay,
+            byte willQos,
+            byte willRetain,
+            int expiryInterval,
+            String16FW contentType,
+            MqttPayloadFormat payloadFormat,
+            String16FW responseTopic,
+            OctetsFW correlationData,
+            OctetsFW willPayload)
+        {
+            final String topic = String.format(SESSION_WILDCARD_TOPIC_FORMAT, clientId);
+            final int topicKey = topicKey(topic);
+
+            final RouteFW route = resolveTarget(routeId, authorization, topic,  PUBLISH_ONLY);
+
+            MqttSessionStateStream stream = sessionStreams.computeIfAbsent(topicKey, s -> new MqttSessionStateStream(0, topic));
+            if (route != null)
+            {
+                final long resolvedId = route.correlationId();
+                stream.initRoute(resolvedId);
```

Comment:
Another possibility is two constructors for `MqttSessionStateStream` instead of `initRoute` where route related fields are also final, with only those needing to be passed being passed to each constructor, depending on `route != null`.